### PR TITLE
Fix --ref for krel ff and enhance documentation

### PR DESCRIPTION
**Pull request by saschagrunert**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

The `--ref` was not working before, which is now correctly taken into
consideration. The documentation around `krel ff` has been improved as
well and small fixes have been applied.

**Which issue(s) this PR fixes**:

None

**Special notes for your reviewer**:

None

**Does this PR introduce a user-facing change?**:

```release-note
- Enhanced `krel ff` documentation and fixed `--ref` to be taken into consideration
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Had to rebase on the latest master branch

**Comment by justaugustus** [comment]
Thanks for the update, @saschagrunert!
Looks good; just needs a rebase.

File: cmd/krel/cmd/ff.go [hunk 1]
```diff
@@ -93,7 +102,7 @@ func runFf(opts *ffOptions) error {
 
 	cleanup := rootOpts.cleanup
 	if cleanup {
-		defer repo.Cleanup() // nolint: errcheck
+		defer func() { err = repo.Cleanup() }()
```
Comment (riking) [inline cmd/krel/cmd/ff.go]: This stomps out an error if repo.Cleanup() succeeds.

Comment (saschagrunert) [reply cmd/krel/cmd/ff.go]: Right, I will revert my change :+1:

**Pull request merged**